.gitignore

```diff
@@ -1,3 +1,5 @@
 /target
 # Ignore Cargo.lock, if this causes trouble, track Cargo.lock in the repo
 Cargo.lock
+
+**/.vscode/settings.json
```

Comment:
Nit.

Why don't you put this in your private gitignore?  The practice is usually for things produced by your editor of choice to be ignored privately by your own settings, not by `.gitignore` that is tracked in the repo.  This `.gitignore` should ignore stuff that will be produced by everyone working with the code, regarldess of the editor.  So in our case, that is stuff produced by the `cargo` tool.